flycheck.el

```diff
@@ -6385,7 +6385,9 @@ This will either be the directory that contains `mix.exs' or,
 if no such file is found in the directory hierarchy, the directory
 of the current file."
   (or
-   (locate-dominating-file (buffer-file-name) "mix.exs")
+   (and
+    buffer-file-name
+    (locate-dominating-file buffer-file-name "mix.exs"))
```

Comment:
I'll repeat my comment from the other PR. Am I wrong in thinking that `buffer-file-name` is a function and it's result should be passed to `locate-dominating-file`?